pallets/utxo/src/lib.rs

```diff
@@ -690,6 +840,31 @@ pub mod pallet {
                     log::debug!("inserting to UtxoStore {:?} as key {:?}", output, hash);
                     <UtxoStore<T>>::insert(hash, Some(output));
                 }
+                Destination::Stake{ stash_pubkey, controller_pubkey, session_key } => {
+                    staking::stake::<T>(stash_pubkey,controller_pubkey,session_key, output.value)?;
+
+                    let hash = tx.outpoint(index as u64);
+                    log::debug!("inserting to LockedUtxos {:?} as key {:?}", output, hash);
+                    <LockedUtxos<T>>::insert(hash, Some(output));
+                    <StakingCount<T>>::insert(controller_pubkey, Some((1,output.value)));
+                }
+                Destination::StakeExtra(controller_pubkey) => {
+                    staking::stake_extra::<T>(controller_pubkey, output.value)?;
+
+                    let hash = tx.outpoint(index as u64);
+                    log::debug!("inserting to LockedUtxos {:?} as key {:?}", output, hash);
+
+                    let (mut num_of_utxos, mut total) = <StakingCount<T>>::get(&controller_pubkey).ok_or("cannot find the public key inside the stakingcount.")?;
+
+                    // update the total locked utxo values
+                    total = total.saturating_add(output.value);
```

Comment:
Honestly, I'm not happy about the `saturating_add` being used everywhere here. If it ever saturates, it can break the invariants of the node and big problems can happen. We should never reach this point. So this (and all similar usages) should be changed to `checked_add` with an `expect()` on the error for the node to crash if this ever happens. This should provably never happen. Currently it's like hoping it won't happen.

Reply:
i'll change them to `checked_add`.